src/Printer.hs

```diff
@@ -0,0 +1,40 @@
+module Printer where
+
+import Data.Tree
+import Data.Maybe
+import Data.Text.Prettyprint.Doc
+
+import Game hiding (Tree(..))
+import Utils
+
+instance Pretty Cell where
+    pretty X = pretty 'x' 
+    pretty O = pretty 'o'
+    pretty B = pretty '.'
+
+instance Pretty Player where
+    pretty PlayerX = pretty ("Player X" :: String)
+    pretty PlayerO = pretty ("Player O" :: String)
+
+prettyRow :: Pretty a => [a] -> Doc ann
+prettyRow r = sep $ zipWith (<+>) (emptyDoc : repeat pipe) (map pretty r)
+
+prettyGrid :: Pretty a => [[a]] -> Doc ann
+prettyGrid g = vsep $ interleave (pretty (replicate dashLength '-')) (map prettyRow g)
+    where 
+        gridSize   = length $ g !! 0
+        dashLength = (gridSize - 1) * 4 + 3
+
+convertTree :: GameTree -> Tree (Grid, Player)
+convertTree (GtNode p ts) = Node p $ map convertTree (catMaybes ts)
+
+prettyTree :: Tree (Grid, Player) -> String
+prettyTree t = drawTree converted
+    where converted = fmap (\(g,p) -> show $ vsep [pretty p, prettyGrid g]) t
+
+prettyGameTree :: Int -> GameTree -> IO () 
```

Comment:
I'd say this function does too many things at once. It's fine for a toy example, but in a bigger project you'd want to separate your code into "library" code with well-defined semantics and "glue"/"app" code where everything goes.

Reply:
Yep, I agree. 
I just wanted a helper function to easily debug `GameTree`s  so will keep this in for now and probably split it up later if usage becomes problematic.